src/shared/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react'
+import Icon from '../Icon'
+import { PaginationWrapper, PaginationButton } from './Pagination.style'
+
+type PaginationProps = {
+  itemsPerPage?: number
+  totalCount?: number
+  currentPage?: number
+  maxPaginationLinks?: number
+  onClick?: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+}
+
+const Pagination: React.FC<PaginationProps> = ({
+  itemsPerPage = 0,
+  totalCount = 0,
+  currentPage = 1,
+  maxPaginationLinks = 5,
+  onClick,
+}) => {
+  const [page, setPage] = useState(currentPage)
```

Comment:
Let's just rely on value from outside, we probably don't need internal state here